freqtrade/rpc/api_server/api_auth.py

```diff
@@ -35,15 +40,47 @@ def get_user_from_token(token, secret_key: str, token_type: str = "access"):
         payload = jwt.decode(token, secret_key, algorithms=[ALGORITHM])
         username: str = payload.get("identity", {}).get('u')
         if username is None:
-            raise credentials_exception
+            if raise_on_error:
+                raise credentials_exception
+            else:
+                return False
         if payload.get("type") != token_type:
-            raise credentials_exception
+            if raise_on_error:
+                raise credentials_exception
+            else:
+                return False
 
     except jwt.PyJWTError:
-        raise credentials_exception
+        if raise_on_error:
+            raise credentials_exception
+        else:
+            return False
     return username
 
 
+# This should be reimplemented to better realign with the existing tools provided
+# by FastAPI regarding API Tokens
+# https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py
+async def get_ws_token(
+    ws: WebSocket,
+    ws_token: Union[str, None] = Query(..., alias="token"),
+    api_config: Dict[str, Any] = Depends(get_api_config)
+):
+    secret_ws_token = api_config.get('ws_token', 'secret_ws_t0ken.')
+    secret_jwt_key = api_config.get('jwt_secret_key', 'super-secret')
+
+    if secrets.compare_digest(secret_ws_token, ws_token):
+        # Just return the token if it matches
+        return ws_token
+    elif user := get_user_from_token(ws_token, secret_jwt_key, raise_on_error=False):
```

Comment:
You can't use the walrus operator - we support python 3.8 still - and will be until it reaches is end of life.

Reply:
I'm confused, we support 3.7? The walrus operator is available in 3.8

Reply:
my confusion - i was thinking it's been intorduced in 3.10 ....

The change needed in `get_user_from_token()` was anyway unacceptable - so the code had to change anyway.